### DISCUSSION TITLE
Dealing with duplicated features

### DISCUSSION THREAD

**Discussion by wudustan** [satijalab/seurat#6002] [Q&A]
In Read10X(), Seurat allows you to make feature names unique by adding .1, .2, .3 etc at the end of duplicated features. However, subsequenctly they are treated as different genes. 

Does it make sense to collapse them into a mean, or take the highest value for each? Some of the features in my data correspond to protein coding genes so this seems like not a trivial issue.

**Comment by xiaowenchenjax** [comment]
I also met this issue. How to deal with duplicate genes? shall we make unique gene set before creating seurat object? any suggestions?
thanks.

**Comment by qslin** [comment]
I have this issue, too. How to find the corresponding ensembl ID?

**Reply by batalha23** [reply]
Hi, I may be a little late to your question but if you look into your features.tsv file (CellRanger output) the first column is EnsemblID and the second is gene name. By default, Seurat's function `Read10x()` uses the second column, but you can call it with `gene.column = 1` to have your transcripts annotated with EnsemblID instead.

**Comment by Hu-Clala** [comment]
Same issue :( 
Is this an issue with the initial GTF file annotation? Does anyone know how to solve it?
